react/Components/inputButtonArea.tsx

```diff
@@ -15,39 +16,90 @@ const InputButtonArea: FC = () => {
     code,
     updateGiftCardFunction,
     setAddValueGiftCard,
+    copyCode,
+    setValidation,
+    loading,
+    loadingCode,
     addValueGiftCard,
   } = useStore()
 
-  function submitFunctionValueButton(e: React.FormEvent<HTMLFormElement>) {
+  function submitFunctionValueButton(e: { preventDefault: () => void }) {
```

Comment:
I think if you use the `React.MouseEvent<HTMLButtonElement>` in this case, is a better ideia
```suggestion
  function submitFunctionValueButton(e: React.MouseEvent<HTMLButtonElement>) {
```